sdk/python/setup.py

```diff
@@ -29,6 +29,13 @@
     "urllib3>=1.15.1",
     "kubernetes>=23.6.0",
     "retrying>=1.3.3",
+    "boto3>=1.33.9",
+    "transformers>=4.20.0",
+    "einops>=0.6.1",
+    "transformers_stream_generator>=0.0.4",
+    "peft>=0.3.0",
+    "datasets>=2.13.2",
+    "huggingface_hub>=0.16.4",
```

Comment:
To avoid adding extra dependencies in the Training Operator SDK, you can define common variables that needs to be used in Storage Init Container and Training Client in `/storage_init_container/types.py`.
Then just re-use `types.py` in Storage Init Container and Training Client.

WDYT @deepanker13 ?

Reply:
I added these dependencies because the GitHub workflow tests fails while importing classes defined in storage_init_containerin training_client.py.
How will adding types.py help with solving the library import issue? @andreyvelich

Reply:
@deepanker13 Why GitHub workflows fails if SDK only requires dependencies from `training_client.py` and `/storage_init_container/types.py` ?

Reply:
In docker files wherever there is pip install -e . , the step fails because of the missing libraries like boto3, transformers etc. They are needed because
of the following import in  training_client.py :-
```
from kubeflow.storage_init_container.s3 import S3DatasetParams
from kubeflow.storage_init_container.hugging_face import (
    HuggingFaceModelParams,
    HuggingFaceTrainParams,
    HfDatasetParams,
    INIT_CONTAINER_MOUNT_PATH,
)
```
@andreyvelich

Reply:
@deepanker13 Are you talking about Dockerfile.conformance ? For me locally the build was successfully.
I think, if you just create `types.py` file in  `storage_init_container` directory and move all required packages there that you mentioned above, you don't need to install all packages in SDK that are required for storage_init_container

Reply:
@andreyvelich with the following types.py
```
from dataclasses import dataclass, field
from urllib.parse import urlparse
import json, os
from datasets import load_dataset
from peft import LoraConfig
import transformers
from transformers import TrainingArguments
import enum
import huggingface_hub
from typing import Union

TRANSFORMER_TYPES = Union[
    transformers.AutoModelForSequenceClassification,
    transformers.AutoModelForTokenClassification,
    transformers.AutoModelForQuestionAnswering,
    transformers.AutoModelForCausalLM,
    transformers.AutoModelForMaskedLM,
    transformers.AutoModelForImageClassification,
]
```
and the following import in hugging_face.py
from kubeflow.storage_init_container.types import *

I am getting this error in conformance test
E   ModuleNotFoundError: No module named 'transformers'

Reply:
@deepanker13 I think, you are getting this error in Conformance:
```
22.43 ERROR: Could not find a version that satisfies the requirement torch>=1.13.1 (from versions: none)
22.43 ERROR: No matching distribution found for torch>=1.13.1
```
That is because torch version is incompatible with HuggingFace dataset.
Please can you remove those decencies from Storage Init Container requirements:
```
torch>=1.13.1
torchvision>=0.9.1
torchaudio>=0.8.1
```
Those libraries should be installed as part of HuggingFace packages: https://github.com/huggingface/datasets/blob/main/setup.py#L183